test/suites/nodeclaim/garbage_collection_test.go

```diff
@@ -147,3 +150,44 @@ var _ = Describe("NodeClaimGarbageCollection", func() {
 		env.EventuallyExpectNotFound(node)
 	})
 })
+
+func ExpectInstanceProfileCreated(instanceProfileName *string) {
+	By("creating an instance profile")
+	createInstanceProfile := &iam.CreateInstanceProfileInput{
+		InstanceProfileName: instanceProfileName,
+		Tags: []*iam.Tag{
+			{
+				Key:   aws.String(coretest.DiscoveryLabel),
+				Value: aws.String(env.ClusterName),
+			},
+		},
+	}
+	By("adding the karpenter role to new instance profile")
+	_, err := env.IAMAPI.CreateInstanceProfile(createInstanceProfile)
+	Expect(ignoreAlreadyExists(err)).ToNot(HaveOccurred())
+	addInstanceProfile := &iam.AddRoleToInstanceProfileInput{
+		InstanceProfileName: instanceProfileName,
+		RoleName:            aws.String(fmt.Sprintf("KarpenterNodeRole-%s", env.ClusterName)),
+	}
+	_, err = env.IAMAPI.AddRoleToInstanceProfile(addInstanceProfile)
+	Expect(ignoreAlreadyContainsRole(err)).ToNot(HaveOccurred())
+}
+
+func ignoreAlreadyExists(err error) error {
+	if err != nil {
+		if strings.Contains(err.Error(), "EntityAlreadyExists") {
```

Comment:
Does IAM already have a constant for this value? Is there room here to convert this to the aws-typed error like we do in the errors package? I'm not against using a string comparison, but it's probably more correct to do an error conversion into the typed thing